### PR TITLE
Update - fix Kraken2 segmentation fault

### PR DESCRIPTION
**Pull request by maxibor**
Fixes the Kraken2 segmentation fault described [here](https://github.com/DerrickWood/kraken2/issues/222).

Solution suggested by @isuruf [here](https://github.com/conda-forge/openmp-feedstock/pull/35)

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by maxibor** [comment]
@BiocondaBot please add label

**Comment by maxibor** [comment]
Any idea with this is failing @Takadonet

**Comment by maxibor** [comment]
Any idea with this is failing @Takadonet @bioconda/core ?
The error on CircleCI:
```
19:53:31 BIOCONDA ERROR COMMAND FAILED (exited with 1): docker run -t --net host --rm -v /tmp/tmpd1955djj/build_script.bash:/opt/build_script.bash -v /home/circleci/project/miniconda/conda-bld/:/opt/host-conda-bld -v /home/circleci/project/recipes/kraken2:/opt/recipe -e LANG=en_US.UTF-8 -e HOST_USER_ID=1001 bioconda/bioconda-utils-build-env:latest /bin/bash /opt/build_script.bash

19:53:31 BIOCONDA ERROR BUILD FAILED recipes/kraken2
19:53:31 BIOCONDA INFO (COMMAND) conda build purge
19:53:32 BIOCONDA ERROR BUILD SUMMARY: of 1 recipes, 1 failed and 0 were skipped. Details of recipes and environments follow.
19:53:32 BIOCONDA ERROR BUILD SUMMARY: FAILED recipe recipes/kraken2
```

**Comment by Takadonet** [comment]
No idea but further up the stack-trace you have lots of weird warnings that I never seen before.

**Comment by maxibor** [comment]
But the only change (besides updating the build number) is adding a runtime dependy:
[`- _openmp_mutex * *_gnu  # [linux]`](https://github.com/bioconda/bioconda-recipes/blob/295ff80cc5fc866ce1df9219c87eb8ae646c705d/recipes/kraken2/meta.yaml#L38), so I don't really get why this is happening.
I was able to build the image locally on a Ubuntu machine without any issues.

**Comment by isuruf** [comment]
Looks like a conda-build bug. Which conda-build version are you using? cc @mbargull

**Comment by maxibor** [comment]
@BiocondaBot please fetch artifacts

**Comment by maxibor** [comment]
Bioconda's CircleCI is using this [Docker image](https://github.com/bioconda/bioconda-utils/blob/master/Dockerfile), with conda 4.8.2

File: recipes/kraken2/meta.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ source:
     - Makefile.patch
 
 build:
-  number: 1
+  number: 2
   has_prefix_files:
     - libexec/kraken2
```
Comment (isuruf) [inline recipes/kraken2/meta.yaml:18]: You can remove the lines 17-20 which triggers the conda-build bug. It's not needed as prefix replacement is done by default

Comment (maxibor) [reply recipes/kraken2/meta.yaml]: Thanks a lot @isuruf, that worked !

**Comment by maxibor** [comment]
@BiocondaBot please add label

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

**Comment by dpryan79** [comment]
@bioconda-bot please merge

Thanks @maxibor and @isuruf !

**Comment by ialbert** [comment]
```
conda create --name test 
conda activate test
conda install kraken2
```

getting a file from taxon benchmarking site:

    wget ftp://ftp.ebi.ac.uk/pub/databases/metagenomics/taxon_benchmarking/B500_V34_200K_1_ocean.fastq.gz

now when run

```
kraken2 -db ~/refs/minikraken_8GB_20200312/ B500_V34_200K_1_ocean.fastq.gz --report out.txt --output out.log
Loading database information... done.
Processed 196400 sequences (49100000 bp) ...Segmentation fault: 11

```

**Pull request merged**